omc3/correction/handler.py

```diff
@@ -76,10 +78,14 @@ def correct(accel_inst: Accelerator, opt: DotDict) -> None:
 
         # ######### Update Model and Response ######### #
         if iteration > 0:
-            LOG.debug("Updating model via MADX.")
+            LOG.debug("Updating model via MAD-X.")
             corr_model_path = opt.output_dir / f"twiss_{iteration}{EXT}"
 
-            corr_model_elements = _create_corrected_model(corr_model_path, [opt.change_params_path], accel_inst)
+            if update_deltap:
+                # update dpp here, as it does not have a MAD-X knob.
+                update_deltap = delta.loc[ORBIT_DPP, DELTA]
```

Comment:
I don't like this.

We define initially `update_deltap` as a bool which is used as a flag to check if we want to do this, but later it takes a value that's passed as parameter.

I would much rather have the flag stay just a flag and make a new variable for the value, which can be `False` otherwise (seems the _create_response... function has this as default value to not update). So this `if` block could be replaced with something like:

```python
varname = delta.loc[ORBIT_DPP, DELTA] if update_deltap else False
```

Reply:
Re: another comment on the `get_update_correction_script` later in my review: this should probably default as `None`, like the argument in that function since it should expect a bool (see comment there).

Reply:
> I don't like this.
> 
> We define initially `update_deltap` as a bool which is used as a flag to check if we want to do this, but later it takes a value that's passed as parameter.
> 
> I would much rather have the flag stay just a flag and make a new variable for the value, which can be `False` otherwise (seems the _create_response... function has this as default value to not update). So this `if` block could be replaced with something like:
> 
> ```python
> varname = delta.loc[ORBIT_DPP, DELTA] if update_deltap else False
> ```

Ahh, sorry, didn't catch this, this is a merge artifact and should not have been pushed.